libs/log-formats.js

```diff
@@ -21,7 +21,27 @@ function prodFormat (type, message, label, namespace) {
   if (message.stack) message = message.stack
 
   // cut the string to avoid a fail in the stack
-  return JSON.stringify({ time: new Date(), type, message, label, namespace }).substr(0, LOG_LENGTH_LIMIT)
+  return JSON.stringify({
+    time: new Date(),
+    type,
+    message: convertCriticalMessage(type, message),
+    label,
+    namespace
+  }).substr(0, LOG_LENGTH_LIMIT)
+}
+
+function convertCriticalMessage(type, message) {
+  if (type !== 'critical') return message
+
+  let result = message
+  if (Array.isArray(result)) {
+    // clean non string items and remove '-' in items
+    result = result
+      .filter(item => typeof item === 'string')
+      .map(item => item.replace(/-/g, ''))
+      .join('-')
+  }
```

Comment:
I am not sure this is the place for this. As stated in #114 , I am not a fan of the API with an Array, I find it clear enough with "LOGIN_FAILED.CAPTCHA". If we keep it simple, we do not have to add logic to the logger to handle this case.

Reply:
I think you are right @ptbrowne . I will let this PR for a moment and close it if no other advice comes.